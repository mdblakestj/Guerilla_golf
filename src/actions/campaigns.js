import uuid from "uuid";
import database from "../firebase/firebase";

//ADD_campaign
export const addCampaign = campaign => ({
  type: "ADD_CAMPAIGN",
  campaign
});

export const startAddCampaign = (campaignData = {}) => {
  return dispatch => {
    const {
      title = "",
      description = "",
      triggerNumber = 0,
      createdBy = "",
      createdAt = 0,
      members = [],
      imageURL = "",
      launched = false,
      emailList = ""
    } = campaignData;
    const campaign = {
      title,
      description,
      triggerNumber,
      createdAt,
      createdBy,
      members,
      imageURL,
      launched,
      emailList
    };
    database
      .ref("campaigns")
      .push(campaign)
      .then(ref => {
        dispatch(
          addCampaign({
            id: ref.key,
            ...campaign
          })
        );
      });
  };
};

//REMOVE
export const removeCampaign = ({ id } = {}) => ({
  type: "REMOVE_CAMPAIGN",
  id
});
//EDIT
export const editCampaign = (id, updates) => ({
  type: "EDIT_CAMPAIGN",
  id,
  updates
});

export const startEditCampaign = (id, updates) => {
  return dispatch => {
    return database
      .ref(`campaigns/${id}`)
      .update(updates)
      .then(() => {
        dispatch(editCampaign(id, updates));
      });
  };
};

export const setCampaigns = campaign => ({
  type: "SET_CAMPAIGNS",
  campaign
});

export const startSetCampaigns = () => {
  return dispatch => {
    return database
      .ref("campaigns")
      .once("value")
      .then(snapshot => {
        const campaigns = [];
        snapshot.forEach(snapshotChild => {
          campaigns.push({
            id: snapshotChild.key,
            ...snapshotChild.val()
          });
        });
        dispatch(setCampaigns(campaigns));
      });
  };
};
