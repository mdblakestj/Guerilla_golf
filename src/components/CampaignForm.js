import React from "react";
import moment from "moment";
import { firebase } from "../firebase/firebase";
import Button from "react-bootstrap/Button";
import { Row, Col } from "react-bootstrap";
import Jumbotron from "react-bootstrap/Jumbotron";
import Form from "react-bootstrap/Form";
import ProgressBar from "react-bootstrap/ProgressBar";

const now = moment();

export default class CampaignForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: props.campaign ? props.campaign.title : "",
      description: props.campaign ? props.campaign.description : "",
      triggerNumber: props.campaign ? props.campaign.triggerNumber : "",
      createdAt: props.campaign ? moment(props.campaign.createdAt) : moment(),
      createdBy: props.campaign ? props.campaign.createdBy : "",
      members: [],
      error: "",
      imageURL: "",
      image: null,
      url: "",
      emailList: "",
      uploadPercentage: 0
    };
    this.fileSelectedHandler = this.fileSelectedHandler.bind(this);
    this.fileUploadHandler = this.fileUploadHandler.bind(this);
  }

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onTitleChange = e => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };
  onTriggerNumberChange = e => {
    const triggerNumber = e.target.value;
    this.setState(() => ({ triggerNumber }));
  };
  onDateChange = createdAt => {
    this.setState(() => ({ createdAt }));
  };

  onSubmit = e => {
    e.preventDefault();
    var user = firebase.auth().currentUser;
    if (
      !this.state.description ||
      !this.state.title ||
      !this.state.triggerNumber
    ) {
      this.setState(() => ({
        error: "Please provide title, description and trigger number"
      }));
    } else {
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        title: this.state.title,
        description: this.state.description,
        createdAt: this.state.createdAt.valueOf(),
        triggerNumber: this.state.triggerNumber,
        createdBy: user.uid,
        members: [user.uid],
        imageURL: this.state.url,
        emailList: user.email
      });
    }
  };
  fileSelectedHandler = e => {
    const image = e.target.files[0];
    this.setState(() => ({ image }));
  };
  fileUploadHandler = () => {
    const { image } = this.state;
    const uploadTask = firebase
      .storage()
      .ref(`images/${image.name}`)
      .put(image);
    firebase
      .storage()
      .ref("images")
      .child(image.name)
      .getDownloadURL()
      .then(url => {
        this.setState(() => ({ url }));
      });

    uploadTask.on(
      "state_changed",
      snapshot => {
        var percentage =
          (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
        this.setState({ uploadPercentage: percentage });
      },
      function error(err) {
        console.log(err);
      },
      function complete() {}
    );
  };

  render() {
    return (
      <div className="form-div">
        {this.state.error && <p>{this.state.error}</p>}

        <div className="campaign-form">
          <h1 className="campaign-h1">Campaign Info</h1>
          <Form onSubmit={this.onSubmit}>
            <Form.Row>
              <Form.Group>
                <input
                  type="text"
                  placeholder="Title"
                  autoFocus
                  value={this.state.title}
                  onChange={this.onTitleChange}
                />
              </Form.Group>
            </Form.Row>
            <Form.Row>
              <Form.Group>
                <input
                  type="number"
                  placeholder="Trigger Number"
                  autoFocus
                  value={this.state.triggerNumber}
                  onChange={this.onTriggerNumberChange}
                />
              </Form.Group>
            </Form.Row>
            <Form.Row>
              <Form.Group>
                <textarea
                  rows="4"
                  cols="50"
                  type="text"
                  placeholder="Description"
                  autoFocus
                  value={this.state.description}
                  onChange={this.onDescriptionChange}
                />
              </Form.Group>
            </Form.Row>
            <button className="btn btn-outline-info" id="google-button">
              Form Submit
            </button>
          </Form>
          <input type="file" onChange={this.fileSelectedHandler} />
          <button
            className="btn btn-outline-info"
            id="google-button"
            onClick={this.fileUploadHandler}
          >
            Upload
          </button>
          <br />
          {this.state.uploadPercentage && this.state.uploadPercentage < 100 && (
            <ProgressBar
              style={{ width: "200px" }}
              now={this.state.uploadPercentage}
            />
          )}

          <img
            src={
              this.state.url ||
              "https://wiki.tripwireinteractive.com/images/4/47/Placeholder.png"
            }
            alt="uploaded images"
            height="300"
            width="400"
          />
        </div>
      </div>
    );
  }
}
