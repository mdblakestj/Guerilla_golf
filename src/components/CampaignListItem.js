import React from "react";
import { connect } from "react-redux";
import { BrowserRouter, Route, Switch, Link, NavLink } from "react-router-dom";
import moment from "moment";
import ProgressBar from "react-bootstrap/ProgressBar";
import {
  Container,
  Row,
  Col,
  Image,
  Card,
  Button,
  Carousel
} from "react-bootstrap";

const CampaignDetails = props => (
  <div className="campaign-title">
    <hr />
    <Row>
      <Col>
        <Image
          src={props.imageURL}
          fluid
          style={{ borderBottom: "10px #357462 solid" }}
        />
      </Col>
      <Col>
        <h3 className="campaign-link">
          <NavLink to={`/campaign/${props.id}`}>{props.title}</NavLink>
        </h3>
        <h4>{props.launched && "  Engine Launched!"}</h4>
        <p>Created: {moment(props.createdAt).format("dddd, MMMM Do YYYY")}</p>
        <h2>{props.description}</h2>
        <p>
          {" "}
          Members: {props.members.length}/{props.triggerNumber}
        </p>

        <ProgressBar
          style={{ width: "200px" }}
          now={(props.members.length / props.triggerNumber) * 100}
        />
      </Col>
    </Row>
  </div>
);

export default CampaignDetails;
