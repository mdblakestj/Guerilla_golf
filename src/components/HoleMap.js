import { connect } from "react-redux";
import React, { Component } from "react";
import { Map, TileLayer, Marker, Popup } from "react-leaflet";
import { Container, Row, Col, Card, CardDeck } from "react-bootstrap";
import CampaignDetails from "./CampaignListItem";

export const CampaignList = props => (
  <div>
    <h1>Holes</h1>

    <Map center={[45, -71]} zoom="7">
      <TileLayer
        attribution='&amp;copy <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
      />
      {props.campaigns.map(campaign => {
        return (
          <Marker position={[campaign.lat, campaign.lng]}>
            <Popup>
              A pretty CSS3 popup. <br /> Easily customizable.
            </Popup>
          </Marker>
        );
      })}
    </Map>
  </div>
);

const mapStateToProps = state => {
  return {
    campaigns: state.campaigns
  };
};

const ConnectedCampaignList = connect(mapStateToProps)(CampaignList);

export default ConnectedCampaignList;
