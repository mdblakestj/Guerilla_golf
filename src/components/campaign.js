import React from "react";
import { connect } from "react-redux";
import { BrowserRouter, Route, Switch, Link, NavLink } from "react-router-dom";
import moment from "moment";
import ProgressBar from "react-bootstrap/ProgressBar";
const Campaign = props => {
  return (
    <div>
      <h1 className="campaign-title"> {props.campaign.title}</h1>
      <h5> {props.campaign.description}</h5>
      <p>Created: {moment(props.createdAt).format("dddd, MMMM Do YYYY")}</p>
      <p>
        {" "}
        Members: {props.campaign.members.length}/{props.campaign.triggerNumber}
      </p>
      <progress
        max="100"
        value={
          (props.campaign.members.length / props.campaign.triggerNumber) * 100
        }
      />
      <p>
        {" "}
        {props.campaign.members.length < props.campaign.triggerNumber
          ? ` ${props.campaign.triggerNumber -
              props.campaign.members.length} more members until engine launched`
          : "Engine Launched!"}{" "}
      </p>
      <br />
      <img src={props.campaign.imageURL} className="img-campaign" />
      <br />
      <br />
      {props.auth.uid && (
        <button
          onClick={() => {
            props.history.push(`/JoinEngine/${props.campaign.id}`);
          }}
        >
          Join Engine!
        </button>
      )}
    </div>
  );
};

// export default Campaign;
const mapStateToProps = (state, props) => {
  return {
    campaign: state.campaigns.find(
      campaign => campaign.id === props.match.params.id
    ),
    auth: state.auth
  };
};

const ConnectedCampaign = connect(mapStateToProps)(Campaign);

export default ConnectedCampaign;
