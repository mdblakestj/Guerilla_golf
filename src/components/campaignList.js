import React from "react";
import { connect } from "react-redux";
import CampaignDetails from "./CampaignListItem";
import getVisibleExpenses from "../selectors/expenses";
import { Container, Row, Col, Card, CardDeck } from "react-bootstrap";
export const CampaignList = props => (
  <div className="campaign-list">
    <Container>
      <Row>
        <h4>Featured Engines </h4>
      </Row>
      {props.campaigns.map(campaign => {
        return (
          <Row key={campaign.id}>
            <CampaignDetails key={campaign.id} {...campaign} />
          </Row>
        );
      })}
      <br />
      <hr />
      <h3>All Engines</h3>
      <CardDeck>
        <Card>
          <Card.Img variant="top" src="holder.js/100px160" />
          <Card.Body>
            <Card.Title>Card title</Card.Title>
            <Card.Text>
              This is a wider card with supporting text below as a natural
              lead-in to additional content. This content is a little bit
              longer.
            </Card.Text>
          </Card.Body>
          <Card.Footer>
            <small className="text-muted">Last updated 3 mins ago</small>
          </Card.Footer>
        </Card>
        <Card>
          <Card.Img variant="top" src="holder.js/100px160" />
          <Card.Body>
            <Card.Title>Card title</Card.Title>
            <Card.Text>
              This card has supporting text below as a natural lead-in to
              additional content.{" "}
            </Card.Text>
          </Card.Body>
          <Card.Footer>
            <small className="text-muted">Last updated 3 mins ago</small>
          </Card.Footer>
        </Card>
        <Card>
          <Card.Img variant="top" src="holder.js/100px160" />
          <Card.Body>
            <Card.Title>Card title</Card.Title>
            <Card.Text>
              This is a wider card with supporting text below as a natural
              lead-in to additional content. This card has even longer content
              than the first to show that equal height action.
            </Card.Text>
          </Card.Body>
          <Card.Footer>
            <small className="text-muted">Last updated 3 mins ago</small>
          </Card.Footer>
        </Card>
      </CardDeck>
      ;
    </Container>
  </div>
);

const mapStateToProps = state => {
  return {
    campaigns: state.campaigns
  };
};

const ConnectedCampaignList = connect(mapStateToProps)(CampaignList);

export default ConnectedCampaignList;
