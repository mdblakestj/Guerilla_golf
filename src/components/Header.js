import React from "react";
import { BrowserRouter, Route, Switch, Link, NavLink } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";
import { Container, Row, Col, Button } from "react-bootstrap";

const Header = props => (
  <header>
    <Container>
      <Row>
        <Col>
          <h1>Guerilla Golf</h1>
        </Col>
        <Col>
          <div className="logout-button">
            {props.auth && (
              <button
                type="button"
                className="btn btn-outline-info"
                onClick={props.startLogout}
              >
                Log Out
              </button>
            )}
          </div>
        </Col>
      </Row>
      <Row>
        <Col>
          <div className="header">
            <NavLink to="/home" activeClassName="is-active" exact={true}>
              Home
            </NavLink>
            <NavLink to="/NewCampaign" activeClassName="is-active">
              New Hole
            </NavLink>
            {props.auth && (
              <NavLink to="/MyCampaigns" activeClassName="is-active">
                My Engines
              </NavLink>
            )}
            {!props.auth && (
              <NavLink to="/" activeClassName="is-active">
                Sign In
              </NavLink>
            )}
          </div>
        </Col>
      </Row>
    </Container>
  </header>
);
const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state, props, dispatch) => {
  return {
    auth: state.auth.uid,
    startLogin: () => dispatch(startLogout())
  };
};

const ConnectedHeader = connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);

export default ConnectedHeader;
