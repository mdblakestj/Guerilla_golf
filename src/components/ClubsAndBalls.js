import React from "react";
import CampaignList from "./campaignList";
import { Container } from "react-bootstrap";
import HoleMap from "./HoleMap";

const ClubsAndBalls = () => (
  <div>
    <HoleMap />
  </div>
);

export default ClubsAndBalls;
