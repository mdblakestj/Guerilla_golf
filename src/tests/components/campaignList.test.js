import React from 'react'
import { shallow } from 'enzyme';
import {CampaignList} from '../../components/campaignList'
import campaigns from '../fixtures/campaigns'

test('should render CampaignList with campaigns', () => {
  const wrapper = shallow(<CampaignList campaigns={campaigns}/>)
  expect(wrapper).toMatchSnapshot();
})
