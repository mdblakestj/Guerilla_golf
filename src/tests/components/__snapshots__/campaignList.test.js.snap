// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CampaignList with campaigns 1`] = `
<div
  className="campaign-list"
>
  <ForwardRef(Bootstrap(Container))>
    <ForwardRef(Bootstrap(Row))>
      <h4>
        Featured Engines 
      </h4>
    </ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Row))>
      <CampaignDetails
        createdAt={100000}
        createdBy="Dan"
        description="HatsHatsHats"
        id="1"
        imageURL="www.test.com"
        key="1"
        members={Array []}
        title="Hats"
        triggerNumber={1000}
      />
    </ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Row))>
      <CampaignDetails
        createdAt={100000}
        createdBy="Dan"
        description="HatsHatsHats2"
        id="2"
        imageURL="www.test.com"
        key="2"
        members={Array []}
        title="Hats2"
        triggerNumber={1000}
      />
    </ForwardRef(Bootstrap(Row))>
  </ForwardRef(Bootstrap(Container))>
</div>
`;
